homeassistant/config_entries.py

```diff
@@ -911,6 +927,48 @@ async def async_step_unignore(self, user_input: Dict[str, Any]) -> Dict[str, Any
         """Rediscover a config entry by it's unique_id."""
         return self.async_abort(reason="not_implemented")
 
+    async def async_step_user(
+        self, user_input: Optional[Dict[str, Any]] = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initiated by the user."""
+        return self.async_abort(reason="not_implemented")
+
+    async def async_handle_discovery_without_unique_id(self) -> None:
```

Comment:
Why make this its own method? Couldn't we just put this in `async_step_discovery`

Reply:
I think we do. See 2 additional example implementation in the PR description.

I rather have people use this instead of things like getting a mac address from the IP address to bend/get around the rules.